Fill.py

```diff
@@ -161,7 +161,13 @@ def fill_restrictive(multiworld: MultiWorld, base_state: CollectionState, locati
             placements.append(spot_to_fill)
             spot_to_fill.event = item_to_place.advancement
             placed += 1
-            if not placed % 1000:
+            if total < 10000 or placed/total < 0.8:
+                if not placed % 1000:
+                    _log_fill_progress(name, placed, total)
+            elif placed/total > 0.95:
+                if not placed % 10:
+                    _log_fill_progress(name, placed, total)
+            elif not placed % 100:
                 _log_fill_progress(name, placed, total)
```

Comment:
It might be nice to put a blank line after this, to make it clearer that the following `if` is intended to be a separate block and not another `elif` in this chain.

Reply:
Added.